payments-ui-core/src/main/java/com/stripe/android/ui/core/FormUI.kt

```diff
@@ -91,32 +95,55 @@ private fun FormUIElement(
     lastTextFieldIdentifier: IdentifierSpec?,
 ) {
     when (element) {
-        is SectionElement -> SectionElementUI(
-            enabled,
-            element,
-            hiddenIdentifiers,
-            lastTextFieldIdentifier
-        )
-        is CheckboxFieldElement -> CheckboxFieldUI(
-            controller = element.controller,
-            enabled = enabled
-        )
+        is SectionElement -> {
+            SectionElementUI(
+                modifier = Modifier.padding(vertical = 8.dp),
+                enabled = enabled,
+                element = element,
+                hiddenIdentifiers = hiddenIdentifiers,
+                lastTextFieldIdentifier = lastTextFieldIdentifier,
+            )
+        }
+        is CheckboxFieldElement -> {
+            CheckboxFieldUI(
+                modifier = Modifier.padding(top = 4.dp),
+                controller = element.controller,
+                enabled = enabled
+            )
+        }
         is StaticTextElement -> StaticTextElementUI(element)
         is SaveForFutureUseElement -> SaveForFutureUseElementUI(enabled, element)
         is AfterpayClearpayHeaderElement -> AfterpayClearpayElementUI(
-            enabled,
-            element
+            enabled = enabled,
+            element = element,
+            modifier = Modifier.padding(4.dp, 8.dp, 4.dp, 4.dp),
+        )
+        is AuBecsDebitMandateTextElement -> AuBecsDebitMandateElementUI(
+            element = element,
+            modifier = Modifier.padding(vertical = 8.dp),
+        )
+        is AffirmHeaderElement -> AffirmElementUI(
+            modifier = Modifier.padding(vertical = 8.dp),
+        )
+        is MandateTextElement -> MandateTextUI(
+            element = element,
+            modifier = Modifier.padding(
+                top = element.topPadding,
+                bottom = 8.dp,
+            )
         )
-        is AuBecsDebitMandateTextElement -> AuBecsDebitMandateElementUI(element)
-        is AffirmHeaderElement -> AffirmElementUI()
-        is MandateTextElement -> MandateTextUI(element)
         is CardDetailsSectionElement -> CardDetailsSectionElementUI(
             enabled,
             element.controller,
             hiddenIdentifiers,
             lastTextFieldIdentifier
         )
-        is BsbElement -> BsbElementUI(enabled, element, lastTextFieldIdentifier)
+        is BsbElement -> BsbElementUI(
+            enabled = enabled,
+            element = element,
+            lastTextFieldIdentifier = lastTextFieldIdentifier,
+            modifier = Modifier.padding(vertical = 8.dp),
+        )
```

Comment:
This PR just removes the individual padding but does not update the form to account for first element and last element padding. That would be followed up in another PR if we think this is the right solution.